Tests/SwiftNavigationTests/ObserveTests/ObserveTests+Nesting.swift

```diff
@@ -0,0 +1,249 @@
+import SwiftNavigation
+import Perception
+import XCTest
+import ConcurrencyExtras
+
+#if !os(WASI)
+  class NestingObserveTests: XCTestCase {
+    @MainActor
+    func testNestedObservationMisuse() async {
+      // ParentObject and ChildObject models
+      // do not use scoped observation in these tests.
+      // This results in redundant updates.
+      // The issue is related to nested unscoped `observe` calls
+      // it is expected behavior for this kind of API misuse
+
+      let object = ParentObject()
+      let model = ParentObject.Model()
+
+      MockTracker.shared.entries.withValue { $0.removeAll() }
+      object.bind(model)
+
+      XCTAssertEqual(
+        MockTracker.shared.entries.withValue { $0.map(\.label) },
+        [
+          "ParentObject.bind",
+          "ParentObject.valueUpdate 0",
+          "ParentObject.value.didSet 0",
+          "ParentObject.childUpdate",
+          "ChildObject.bind",
+          "ChildObject.valueUpdate 0",
+          "ChildObject.value.didSet 0",
+        ]
+      )
+
+      MockTracker.shared.entries.withValue { $0.removeAll() }
+      model.child.value = 1
+
+      // Those are not enough to perform updates:
+      // await Task.yeild()
+      // await Task.megaYeild()
+      // Falling back to Task.sleep
+      try? await Task.sleep(nanoseconds: UInt64(0.5 * pow(10, 9)))
+
+      XCTAssertEqual(
+        MockTracker.shared.entries.withValue(\.count),
+        13
+      )
```

Comment:
The order of logs is not consistent here, so I decided to assert by the count which seems stable across runs